elide-datastore/elide-datastore-search/src/main/java/com/yahoo/elide/datastores/search/SearchDataTransaction.java

```diff
@@ -0,0 +1,252 @@
+    public Iterable<Object> loadObjects(Class<?> entityClass,
+                                        Optional<FilterExpression> filterExpression,
+                                        Optional<Sorting> sorting,
+                                        Optional<Pagination> pagination,
+                                        RequestScope requestScope) {
+
+        if (!filterExpression.isPresent()) {
+            return super.loadObjects(entityClass, filterExpression, sorting, pagination, requestScope);
+        }
+
+        boolean canSearch = canSearch(filterExpression.get(), entityClass);
+
+        if (mustSort(sorting, entityClass)) {
+            canSearch = canSearch && canSort(sorting.get(), entityClass);
+        }
+
+        if (canSearch) {
+            return search(entityClass, filterExpression.get(), sorting, pagination);
+        }
+
+        return super.loadObjects(entityClass, filterExpression, sorting, pagination, requestScope);
+    }
+
+    /**
+     * Indicates whether sorting has been requested for this entity
+     * @param sorting An optional elide sorting clause.
+     * @param entityClass The entity to sort.
+     * @return True if the entity must be sorted. False otherwise.
+     */
+    private boolean mustSort(Optional<Sorting> sorting, Class<?> entityClass) {
+        return sorting.isPresent() && !sorting.get().getValidSortingRules(entityClass, dictionary).isEmpty();
+    }
+
+    /**
+     * Returns whether or not Lucene can be used to sort the query.
+     * @param sorting The elide sorting clause
+     * @param entityClass The entity being sorted.
+     * @return true if Lucene can sort.  False otherwise.
+     */
+    private boolean canSort(Sorting sorting, Class<?> entityClass) {
+
+        for (Map.Entry<Path, Sorting.SortOrder> entry
+                : sorting.getValidSortingRules(entityClass, dictionary).entrySet()) {
+
+            Path path = entry.getKey();
+
+            if (path.getPathElements().size() != 1) {
+                return false;
+            }
+
+            Path.PathElement last = path.lastElement().get();
+            String fieldName = last.getFieldName();
+
+            if (dictionary.getAttributeOrRelationAnnotation(entityClass, SortableField.class, fieldName) == null) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Verifies whether or not a filter expression can be searched with Lucene.
+     * @param expression An elide filter expression.
+     * @param entityClass The entity being searched.
+     * @return true if Lucene can be used.  False otherwise.
+     */
+    private boolean canSearch(FilterExpression expression, Class<?> entityClass) {
+        Collection<FilterPredicate> predicates = expression.accept(new PredicateExtractionVisitor());
+
+        return predicates.stream().allMatch((predicate) -> {
+            if (predicate.getPath().getPathElements().size() != 1) {
+                return false;
```

Comment:
What would make a multi-entity path unsearchable?

Reply:
Multiple path elements means we have to join to other tables.